registry/repositories/user.postgres.ts

```diff
@@ -0,0 +1,53 @@
+import { QueryConfig } from "pg";
+import { pgPool } from "@/repositories/connection.postgres.js";
+import {
+  UserRepository,
+  User,
+  AuthBasicSignup,
+} from "@/repositories/user.interface.js";
+
+export const createUserRepository = (): UserRepository => {
+  return {
+    async getUser(username: string): Promise<User | null> {
+      const query: QueryConfig = {
+        name: "queryGetUserByUsername",
+        text: `
+          SELECT user_id, username, password, created_at
+          FROM users 
+          WHERE username = $1;
+        `,
+        values: [username],
+      };
+      const result = await pgPool.query<User>(query);
+      return result.rows.length ? result.rows[0] : null;
```

Comment:
```suggestion
      return result.rows.at(0)
```
or this if null should be returned if the user does not exist.
```suggestion
      return result.rows.at(0) ?? null;
```